### PR TITLE
Feat/hybrid

### PR DESCRIPTION
**Pull request by Nesopie**
This PR: 
- adds hybrid support to cjs and esm modules
- moves buffer usage into Uint8Array

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "extends": "./tsconfig.base.json",
+    "compilerOptions": {
+      "outDir": "src/esm",
+      "esModuleInterop": true,
```
Comment (junderw) [inline tsconfig.json]: esModuleInterop is only useful for commonjs

ESM does not need interop considerations with ESM.

File: tsconfig.base.json [hunk 2]
```diff
@@ -0,0 +1,24 @@
+{
+    "compilerOptions": {
+      "target": "ES5",
+      "module": "commonjs",
+      "outDir": "./src",
```
Comment (junderw) [inline tsconfig.base.json]: module and outDir are both defined in the other configs. Not really needed.

File: tsconfig.json (hunk 1 above)
Comment (Nesopie) [reply tsconfig.json]: you're right..., will move it

**Comment by junderw** [comment]
Can you write a benchmark that compares all these DataView and ArrayBuffers to something simpler?

I think we should add some bit writing functions in uint8array-tool and use those.

But if DataView and ArrayBuffer are fast enough it might be fine.

File: src/esm/index.js [hunk 3]
```diff
@@ -46,29 +56,33 @@ export function decode(buffer, offset) {
...
+            numberValue: val,
+            bigintValue: BigInt(val),
             bytes: 5
         };
         // 64 bit
     }
     else {
         const number = tools.readUInt64(buffer, offset + 1, 'LE');
-        return { value: number, bytes: 9 };
+        return { numberValue: null, bigintValue: number, bytes: 9 };
```
Comment (junderw) [inline src/esm/index.js]: if number is `<=` to `Number.MAX_SAFE_INTEGER` then return the numberValue.

**Review by junderw — APPROVED** [review]
ACK 95ff37a90a6ac64a977660689b799a3b7e37f1d3

**Pull request merged**